### ISSUE TITLE
I can't change the audio track

### ISSUE THREAD

**Issue by Joab0** [jellyfin/jellyfin-android#1502]
### Describe the bug

I'm having trouble switching between audio tracks on some content. On the web, I can easily switch between tracks without any issues, but on the Android app (using the default Web player), the audio track doesn't change.

To clarify, the content has audio tracks in both pt-BR and en-US. However, even when I select the en-US track, the playback remains in pt-BR.

### Application version

2.6.2

### Where did you install the app from?

None

### Device information

Xiaomi Redmi Note 12

### Android version

Android 14

### Jellyfin server version

10.9.11

### Which video player implementations does this bug apply to?

- [X] Web player (default)
- [ ] Integrated player (ExoPlayer)
- [ ] External player (VLC, mpv, MX Player)

**Comment by jetzzze** [comment]
I experience the same problem on two different smart phones

Application version
2.6.2

Where did you install the app from?
None

Device information
Samung Galaxy S22 and S24

Android version
Android 13 and 14

Jellyfin server version
10.10.0

Which video player implementations does this bug apply to?
The Android app internal player

Thank you for considering this as a bug and for fixing it 👍

**Comment by tcely** [comment]
I also ran into this with a OTA recording where I put the main and SAP audio streams into the mkv.

The web player says I am playing the SAP Audio Description stream (it even reports it as stereo instead of the 6 channels main has) but the actual audio I hear doesn't change (the audio descriptions aren't playing).

**Comment by PommesMitFritten** [comment]
Same issue here, although with a twist: Without subtitles, picking a different audio track has no effect, the movie continues to play in the default language/English. But if I choose any subtitles, the audio changes to the selected track.

This issue does not occur using the web interface in the mobile google chrome browser app, neither for the "normal" player nor for the "embedded" player.

Android 14
Jellyfin App 2.6.2
Jellyfin Server 10.10.3

**Comment by karlicoss** [comment]
Switching to "Integrated player" solved the same issue for me!
In addition, seems like it fixed issue when Jellyfin was eating 400% CPU transcoding MKV movies with subtitles (mediainfo said `Picture based subtitle format used on BDs/HD-DVDs`). Looks like now it's all via direct play

**Comment by BotBlake** [comment]
I also cannot reproduce this issue.
Can someone give me a heads up if this is still an issue?

Kind regards

**Comment by jetzzze** [comment]
Hi,
I wanted to test with 
Jelly client app Android version: 2.6.3-beta1
and server version: 10.10.7

but me phone just upgraded to Android 18 One-Ui 7.0 and Jellyfin app does not work ... need to fix this first :-) 

I tested with 
Jelly client app Android version: 2.6.3-beta1
and server version: 10.10.1

and the problem was still there

**Comment by BotBlake** [comment]
I'm on Android 15 OneUI 7 and do not have any issues.
I am not aware of any issues with the Jellyfin App and OneUI 7 😅

**Comment by jetzzze** [comment]
Ok, need to correct my findings. I tested with a different set of audio files and it seems not generally a problem anymore.

It works like 50:50 ... I have files where it now switches the audio language but I have others where it does not.

BUT when playing those files via the web UI directly from the Jellyfin server the switch works flawlessly.
So the second audio track is working but not in the app.

But I fear it will be hard to chase as some files are working and some are not ...

I can provide a file for testing if wanted :-)

**Comment by tcely** [comment]
> I found the reason!
> 
> When the file is in **MP4** format the audio track change **does not work**. If it is an **MKV** container then it **works** !!!
> 

I don't think that is going to be the reason. I was using `.mkv` files when I saw the problem.

> I converted the MP4 file that did not work with Handbrake into a MKV container format and then it worked.

Does your converted file still have multiple audio tracks? Are they still in the same order? Was a default flag added?

**Comment by jetzzze** [comment]
> Does your converted file still have multiple audio streams? Are they still in the same order? Was a default flag added?

Yes I have assigned the two existing audio tracks (AAC 5.1 German and AAC 5.1 English) in the same order to the MKV file.
I am not aware of setting a default flag with Handbrake. And I definitely did not explicitly set one.

If you say it can not be the container format I will try to check the metadata info of the video files. Can you advise what to look for?

**Comment by jetzzze** [comment]
I found a difference the **_working file (in my case mkv)_** has a "Forced" flag set to "No":
`Language                                 : German
Default                                  : Yes
Forced                                   : No`

while the not working file (in my case mp4) does not even have the "Forced" flag:
`Language                                 : German
Default                                  : Yes`

Has this something to do with the default flag?
Can it set this flag manually into the metadata?

**Comment by jetzzze** [comment]
After checking the web player in the browser I have now checked how the Jellyfin app on my Android TV (jellyfin-androidtv 0.18.8) and on my Fire TV stick (0.18.8 as well) behaves. They are switching the audio track when selecting the second (Non-Standard) audio track. So it the issue can be narrowed down to the Android phone/tablet app.

Let me know if I can do any further testing or collecting metadata info from my video files.

Cheers

**Comment by YoannPa** [comment]
I got the exact same issue on a OnePlus 13R.
It is the current version of the Android Jellyfin App.

Will try the solution suggested by @PommesMitFritten

**Comment by YoannPa** [comment]
I can confirm that the solution proposed by @PommesMitFritten worked on OnePlus 13R : 

> Switching "Settings->Client Settings->Video player type" from "Web player" to "Integrated player" in the app, solves the issue.

**Comment by jetzzze** [comment]
> Edit: Switching "Settings->Client Settings->Video player type" from "Web player" to "Integrated player" in the app, solves the issue.

I can also confirm that this workaround works. It is up to the developers to decide if the webplayer behaviour needs to be fixed or if the workaround is considered the solution for this issue :-)

**Comment by marschlu** [comment]
Issue still persist

**Comment by jogicom** [comment]
I have also observed the following behavior (WebPlayer):

Subtitles **enabled**: audio selection **works**

Subtitles **disabled**: The **first** available audio track is always played, regardless of which audio track is selected.

**Comment by Fr0zka** [comment]
I wish it was solved owo sadly on webplayer, whatever i tried to do ? on direct play it select my system languague, so french :c no modification anywhere even on file for the "default audio track" can solve it </3

**Comment by bedbug1226** [comment]
I am also having this issue and this workaround fixed it.

> Edit: Switching "Settings->Client Settings->Video player type" from "Web player" to "Integrated player" in the app, solves the issue.

**Comment by n3-rf** [comment]
Issue is still present !

**Comment by PlayerXYT** [comment]
This is a problem for me too.

Server: Linux 7.0.11.arch-1 (No docker) @ JF 10.11.10-1
Client 1: Lenovo tab M11 @ Android 14, JF 2.6.4 F-Droid
Client 2: Google pixel 6a @ Android 16, JF 2.6.4 F-Droid

Trying to switch audio from my PC (the """server""") on firefox works fine, both mobile clients with the web player have this bug. On my phone using the integrated player or VLC it works as expected (untested on my tablet).

**Comment by PoweredTable** [comment]
The issue persists

**Comment by jfrykman** [comment]
I'm also having this issue with some files. As another commenter pointed out it seems to only be an issue when subtitles are being displayed. What seems to be happening is that having subtitles enabled is causing the video to be transcoded. This is working fine for my audio tracks that are 2.0 AAC, but when I try to select an audio track which is 5.1 AAC (or AC-3) it switches back to one of the AAC 2.0 tracks after starting the transcode process. 

For this file in particular I have 3 Japanese tracks in 2.0 AAC, 1 English track in 2.0 Dual-Mono AAC, 2 tracks in English 5.1 AAC and 1 track in English 5.1 AC-3. All the Japanese tracks work, but if I select any of the English 5.1 tracks it starts transcoding and switches back to the English 2.0 Dual-Mono track. Hopefully this helps track down the issue.
